Add onSwatchHover to CompactPicker

pconerly opened the pull request:
@casesandberg 

For #365

I only added `onSwatchHover` for `CompactPicker`, but if you'd like me to add it to any picker that wraps Swatch, I'd be happy to. And in general, I'm open to any ideas for getting this functionality in.

casesandberg commented:
This looks stellar! Would you mind adding it for all of the pickers?

casesandberg commented on package.json:
```diff
@@ -22,7 +22,8 @@
     "clean": "rm -rf lib && mkdir lib",
     "lib": "npm run clean && babel src -d lib",
     "docs": "node ./scripts/docs-server",
```
Nit-picky, can you change the base "docs" command to `npm run docs-server` so that we don't have duplicate command code in there?

pconerly replied:
Done!

pconerly commented:
@casesandberg so I added it to all the relevant higher-level components (`Block`, `Circle`, `Compact`, `Github`, `Sketch`, `Swatches`, `Twitter`)

I also added tests to test the eventing. I struggled doing it with the existing test utils, so I added airbnb's `enzyme`. If that's a no-go LMK and I can try to figure out another way, but I think enzyme is really great.


I also found that I had unnecessary code in `Compact` where I was making the `optionalEvents` object that was splatted into it.  Only `Swatch` needs to do the null check, all of the other components can happily pass `null`s around.

casesandberg commented on package.json:
```diff
@@ -17,11 +17,12 @@
     "clean": "rm -rf lib && mkdir lib",
     "lib": "npm run clean && babel src -d lib",
-    "docs": "node ./scripts/docs-server",
+    "docs-server": "node ./scripts/docs-server",
```
Ah, sorry, should have been more clear here. Please leave both `docs` and `docs-server` but have the docs script just call the docs-server one: 
```
"docs": "npm run docs-server",
```

casesandberg commented:
This is wonderful! It has been on my todo to add more comprehensive tests with enzyme, so this is perfect! Just one last thing with the run scripts and I can merge this in!

pconerly commented:
Great to hear :) I'll fix the package.json thing.

I'm also not sure if eslint is working properly? I'm not getting any errors, but the `eslintrc.js` references ` extends: '@case/eslint-config' ` ? And I'm not sure how that works.

casesandberg commented:
Eslint is working-- You wrote the code with the right formatting, so there are no errors!

casesandberg commented:
This is live in `react-color@2.13.0` 🎉  Thanks for such a great first contribution!

pconerly commented:
Thanks for your help too :) It was easy to add this change in such a well-organized project!
